ds-hashtable/map-query.ts

```diff
@@ -0,0 +1,16 @@
+import { Document, DocumentIndex } from './map-index';
+
+/**
+ * Queries a DocumentIndex and returns all the Documents that contain
+ * any of the words in the query, as a Set.
+ */
+export function queryIndex(index: DocumentIndex, query: string): Set<Document> {
+  const value = new Set<Document>();
+  const words = query.match(/\b(\w+)\b/g) || [];
+  for (const word of words) {
+    const docs = index.get(word.toLowerCase());
+    if (docs) {
```

Comment:
In here, just loop through the docs and add each one to `value`. Something like:
```
for (const doc of docs) {
  value.add(doc);
}
```